### PR TITLE
A00 - Expose error api (except for src/git, more involved)

### PR DESCRIPTION
**Pull request by blaggacao**

### CODE REVIEW

**Comment by blaggacao** [comment]
@kevgo not 100% sure about the wording, as I learned late (through failing feature tests) that those are habitually user facing... Exposing a proper error API throughout everything might be something for another day, for now, i just tried to get rid o the `goerr113` linter errors...

File: src/cmd/kill.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func getKillConfig(args []string, repo *git.ProdRepo) (result killConfig, err er
 		result.targetBranch = args[0]
 	}
 	if !repo.IsFeatureBranch(result.targetBranch) {
-		return result, fmt.Errorf("you can only kill feature branches")
+		return result, fmt.Errorf("cannot be killed: %w", ErrNoFeatureBranch)
```
Comment (charlierudolph) [inline src/cmd/kill.go:68]: Thoughts on this being: `branch %q: not a feature branch`

**Comment by blaggacao** [comment]
@charlierudolph addressed your issue, should error declarations probably go into `error.go` files ...

**Comment by kevgo** [comment]
Thanks for fixing this @blaggacao! I actually don't agree with the `goerr113` linter here. Its opinion is correct when writing libraries, but we are writing a CLI tool here (that doesn't expose an API for now) and therefore don't have clients that need to match on error types. Hence, creating so many sentinel errors feels like unnecessary boilerplate. I have "fixed" this by disabling that particular linter check in #1496. More details at https://github.com/Djarvur/go-err113/issues/10.

When we add an API for Git Town later, we need to design the error types for long-term stability. I'm thinking something along the lines of `UserError` for things where the user did something wrong (and the caller should try again with better input) and `InternalError` for bugs in Git Town. But as you say, this is something for another day. 

I prefer that we keep the user-visible error messages as human friendly as possible. Hence I'm going to close this PR, since #1496 fixes this in a different way that doesn't change the user-visible behavior. Sorry again for running into these issues with all these brand new linters, and thanks again for fixing this!

**Comment by blaggacao** [comment]
Totally agree, it felt somewhat linter-driven and short of the root task, anyway.

**Pull request closed without merging**